### PR TITLE
fixes issue #36

### PR DESCRIPTION
**Pull request by codingdave**
This fix uses a support file to pass to include-what-you-use.exe. Long
argument lists happen to not be supported.

### CODE REVIEW

**Comment by codingdave** [comment]
Hi, I went into the same problem like discussed in #36 and fixed it using support files.

File: IncludeToolbox/IncludeWhatYouUse/IWYU.cs [hunk 1]
```diff
@@ -263,64 +256,69 @@ static public string RunIncludeWhatYouUse(string fullFileName, EnvDTE.Project pr
...
-                    // Set max line length so something large so we don't loose comment information.
-                    // Documentation:
-                    // --max_line_length: maximum line length for includes. Note that this only affects comments and alignment thereof,
-                    // the maximum line length can still be exceeded with long file names(default: 80).
-                    process.StartInfo.Arguments += "-Xiwyu --max_line_length=1024 ";
-
-                    // Custom stuff.
-                    process.StartInfo.Arguments += settings.AdditionalParameters;
-                    process.StartInfo.Arguments += " ";
+                    settings.MappingFiles = new string[] { "stl.c.headers.imp", "msvc.imp", "boost-all.imp", "boost-all-private.imp" };
```
Comment (Wumpf) [inline IncludeToolbox/IncludeWhatYouUse/IWYU.cs:280]: Why are you manipulating the settings outside of the settings screen?
Also, much more important, where do you get this list from? Not having those files will result in errors while running the tool. On top of that for me only absolute paths work which is why the automatic download puts the filenames as such into the settings.

File: IncludeToolbox/IncludeWhatYouUse/IWYU.cs [hunk 2]
```diff
@@ -263,64 +256,69 @@ static public string RunIncludeWhatYouUse(string fullFileName, EnvDTE.Project pr
...
+                var defines = " -D " + string.Join(" -D", preprocessorDefintions.Split(';'));
 
-                // Finally, the file itself.
-                process.StartInfo.Arguments += "\"";
-                process.StartInfo.Arguments += fullFileName;
-                process.StartInfo.Arguments += "\"";
+                // write support file. Long argument lists lead to an error. Support files are the solution here.
+                // https://github.com/Wumpf/IncludeToolbox/issues/36
+                var supportFile = Path.GetTempFileName();
+                var supportContent = includes + " " + defines + " " + Regex.Escape(fullFileName);
```
Comment (Wumpf) [inline IncludeToolbox/IncludeWhatYouUse/IWYU.cs:318]: I don't think Regex.Escape is what you want here. As I understand it this is for escaping regex metacharacters so they are matched as characters. but fullFileName is clearly not a regex. So just put it in quotes as it was before.

**Comment by Wumpf** [comment]
didn't think of trying support/response files to fix those issues, didn't look like too long command line to me. But if that solves it, sure 🤷‍♀️
Please address the mentioned issues and come up with a better commit message ;)

File: IncludeToolbox/IncludeWhatYouUse/IWYU.cs (hunk 1 above)
Comment (codingdave) [reply IncludeToolbox/IncludeWhatYouUse/IWYU.cs:280]: Good spot. There is no need. I will fix that.

File: IncludeToolbox/IncludeWhatYouUse/IWYU.cs (hunk 2 above)
Comment (codingdave) [reply IncludeToolbox/IncludeWhatYouUse/IWYU.cs:318]: You are right. I will have a look on this soon and fix the PR.

**Comment by codingdave** [comment]
For us this reliably solves the issue. I may even want to promote some more flags for use of VS. It would be also very interesting for us if someone had a msvc imp file (the one I downloaded on the IWYU github does not do a good job for us).

I dont understand how I can improve my commit message. Any guidelines for that?

Kind regards

**Comment by codingdave** [comment]
I have created a new PR #60 with a reworked commit, hopefully better commit message, no Regex.Escape and still using support files for fixing the bug many users have.

I would be happy for any kind of feedback.

Cheers,
Dave

**Pull request closed without merging**